python/LAMBDA_DLQ_CHECK/LAMBDA_DLQ_CHECK.py

```diff
@@ -0,0 +1,434 @@
+
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+'''
+
+#####################################
+##           Gherkin               ##
+#####################################
+
+Rule Name: LAMBDA_DLQ_CHECK
+
+Description: Checks whether the AWS Lambda function is configured for a Dead Letter Queue(DLQ). The rule is NON_COMPLIANT if the Lambda function is not configured for DLQ. Trigger: Periodic
+
+Trigger:
+  Configuration Change
+
+Reports on:
+  AWS::Lambda::Function
+
+Rule Parameters:
+  dlqArn
+  (Optional) Comma-separated list of SQS and SNS ARN that Lambda function must be configured for as a DLQ
+
+Scenarios:
+  Scenario: 1
+     Given: The dlqArn rule parameter is not valid (not a valid SNS/SQS ARN)
+      Then: Return Error
+  Scenario: 2
+     Given: Lambda function is not configured for DLQ i.e. DeadLetterConfig is not present
+      Then: Return NON_COMPLIANT with Annotation "This Lambda function is not configured for DLQ"
+  Scenario: 3
+     Given: Lambda function is configured for DLQ i.e. DeadLetterConfig is present
+       And: The rule parameter dlqArn is empty
+      Then: Return COMPLIANT
+  Scenario: 4
+      Given: Lambda function is configured for DLQ i.e. DeadLetterConfig is present
+        And: No ARN in the dlqArn rule parameter doesn't match with the DeadLetterConfig ARN

```

Comment:
Replace the double negation `No ARN in the dlqArn rule parameter matches with the DeadLetterConfig ARN`

Reply:
Done